### PR TITLE
feat: Adding ckERC20 pages and chain-key token requirements

### PR DESCRIPTION
**Pull request by THLO**
The PR adds documentation about ckERC20 tokens under `Chain Fusion > Chain-key tokens` with the same subpages as for ckBTC and ckETH.

Apart from a few minor improvements to the ckETH pages, the PR also specifies the requirements for chain-key tokens on the "Chain-key token overview" page.

### CODE REVIEW

File: docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx [hunk 1]
```diff
@@ -19,6 +19,18 @@ Chain-key tokens can be swapped for one another seamlessly. Making transactions
 
 Each implementation of a chain-key token uses the [ICRC](https://github.com/dfinity/ICRC) token standard so that they can be easily integrated by services such as wallets and DEXs. To implement smart contracts that use chain-key tokens, simply make calls from a smart contract to the ICRC ledger for the chain-key token you wish to use. Using chain-key tokens does not require implementing a specific type of smart contract, and implementation is not restricted to a specific programming language.
 
+## Requirements for chain-key tokens
+
+In addition to the requirement that the chain-key token adheres to the  [ICRC](https://github.com/dfinity/ICRC) token standard, there are additional requirements to ensure a high degree of security and decentralization for all chain-key tokens:
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx:24]:
```suggestion
In addition to the requirement that a chain-key token must adhere to the [ICRC](https://github.com/dfinity/ICRC) token standard, there are additional requirements to ensure a high degree of security and decentralization for all chain-key tokens:
```

File: docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx [hunk 2]
```diff
@@ -19,6 +19,18 @@ Chain-key tokens can be swapped for one another seamlessly. Making transactions
...
+
+- The chain-key token must be a true digital "twin" in the sense that any user can convert the native token to the chain-key token and vice versa, paying only for the fees. This implies that the chain-key token must be fully backed by the corresponding native token at all times. Conversions may be delayed if one chain is temporarily unavailable but all conversions must be finalized eventually.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx:26]: > paying only for the fees.

What fees does this refer to, the transfer fees?

File: docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx [hunk 3]
```diff
@@ -19,6 +19,18 @@ Chain-key tokens can be swapped for one another seamlessly. Making transactions
...
+
+- All conversions must be transparent, that is, it must be possible to trace all tokens on chain.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx:28]:
```suggestion
- All conversions must be transparent. It must be possible to trace all tokens on-chain.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx [hunk 4]
```diff
@@ -19,6 +19,18 @@ Chain-key tokens can be swapped for one another seamlessly. Making transactions
...
+
+- The canister handling the conversions must use chain-key signing capabilities to manage the underlying native token. Moreover, it must read data from the native chain in a decentralized manner.
+
+- All canisters must be decentralized, which means that all source code must be open, builds must be reproducible, and the canisters must be immmutable or controlled by a DAO (for example, an [SNS](https://internetcomputer.org/docs/current/developer-docs/daos/sns) or a [threshold canister](https://github.com/dfinity/threshold) with a large number of verified voters).
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx:32]:
```suggestion
- All canisters must be decentralized, meaning all source code must be open, builds must be reproducible, and the canisters must be immutable or controlled by a DAO (for example, an [SNS](https://internetcomputer.org/docs/current/developer-docs/daos/sns) or a [threshold canister](https://github.com/dfinity/threshold) with a large number of verified voters).
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+# ckERC20 overview
+
+<MarkdownChipRow labels={["Advanced", "Ethereum" ]} />
+
+## What is ckERC20?
+
+Chain-key ERC-20 (ckERC20) tokens are
+[ICRC-2](https://github.com/dfinity/ICRC-1/blob/main/standards/ICRC-2/README.md)-compliant tokens that are backed 1:1 by the corresponding [ERC-20](https://ethereum.org/en/developers/docs/standards/tokens/erc-20/)-compliant token on Ethereum.
+
+Just like other chain-key tokens, such as [ckBTC](/docs/current/developer-docs/multi-chain/chain-key-tokens/ckbtc/overview) and [ckETH](/docs/current/developer-docs/multi-chain/chain-key-tokens/cketh/overview), the ckERC20 tokens do not rely on any centralized bridge to facilitate the conversion between the ckERC20 token and the ERC-20 tokenETH and ckETH, which makes the conversion substantially more secure compared to other traditional 'wrapped' tokens.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:17]:
```suggestion
Just like other chain-key tokens, such as [ckBTC](/docs/current/developer-docs/multi-chain/chain-key-tokens/ckbtc/overview) and [ckETH](/docs/current/developer-docs/multi-chain/chain-key-tokens/cketh/overview), the ckERC20 tokens do not rely on any centralized bridge to facilitate the conversion between the ckERC20 token and the ERC-20 token, which makes the conversion substantially more secure compared to other traditional 'wrapped' tokens.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+Moreover, like ckETH, ckERC20 tokens offer the same advantages over their counterparts on Ethereum, which are faster and cheaper transactions.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:19]:
```suggestion
ckERC20 tokens offer the same advantages over their counterparts on Ethereum, including faster and cheaper transactions.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+A ckERC20 token can be created for any ERC-20 token. The process of adding a ckERC20 token involves an NNS proposal as described [here](https://github.com/dfinity/ic/blob/master/rs/ethereum/ledger-suite-orchestrator/README.adoc#add-new-ckerc20).
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:21]:
```suggestion
A ckERC20 token can be created for any ERC-20 token. The process of adding a ckERC20 token involves creating an [NNS proposal](https://github.com/dfinity/ic/blob/master/rs/ethereum/ledger-suite-orchestrator/README.adoc#add-new-ckerc20).
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+- Spawn a new ICRC ledger suite when adding a new ckERC20 token via an NNS proposal.
+
+- Upgrade all managed ledger suites via an NNS proposal.
+
+- Top up the canisters of all managed ledger suites with cycles.
+
+In order to convert ERC-20 tokens to the corresponding ckERC20 tokens, the following steps are carried out:
+
+1. The user submits an Ethereum transaction calling the `approve` function of the ERC-20 smart contract to allow the helper smart contract to withdraw some the user's funds.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:50]:
```suggestion
1. The user submits an Ethereum transaction calling the `approve` function of the ERC-20 smart contract to allow the helper smart contract to withdraw some of the user's funds.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+The ledger suite orchestrator is responsible for a set of _ICRC ledger suites_, one ledger suite per ckERC20 token. Each ledger suite consists of a ledger, an index, and possibly one or more archive canisters.
+The ledger suite orchestrator makes it possible to handle all ckERC20 tokens in a scalable manner.
+Specifically, it can perform the following actions:
+
+- Spawn a new ICRC ledger suite when adding a new ckERC20 token via an NNS proposal.
+
+- Upgrade all managed ledger suites via an NNS proposal.
+
+- Top up the canisters of all managed ledger suites with cycles.
+
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:47]:
```suggestion
## Converting ERC-20 tokens to ckERC20 tokens
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+In order to convert ERC-20 tokens to the corresponding ckERC20 tokens, the following steps are carried out:
+
+1. The user submits an Ethereum transaction calling the `approve` function of the ERC-20 smart contract to allow the helper smart contract to withdraw some the user's funds.
+
+2. The user calls the `deposit` function of the helper smart contract, specifying the ERC-20 contract, how many ERC-20 tokens should be withdrawn from the user's account, and the principal ID that should be credited for the minted ckERC20 tokens. The helper smart contract transfers the specified number of ERC-20 tokens to itself and, if successful, emits an `event` to be consumed by the ckETH minter.
+
+3. The ckETH minter periodically fetches the logs from the helper smart contract, minting ckERC20 tokens for every newly discovered `event`.
+
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:55]:
```suggestion
## Converting ckERC20 tokens to ERC-20 tokens
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+A user can convert ckERC20 tokens to ERC-20 tokens by following these steps:
+
+1. The user calls the `icrc2_approve` function on the ckETH ledger to approve the ckETH minter to burn some of the user's ckETH tokens as payment for the transaction fees.
+
+2. The user calls the `icrc2_approve` function on the ledger of the given ckERC20 token to approve the minter to burn some of the user' ckERC20 tokens.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:60]:
```suggestion
2. The user calls the `icrc2_approve` function on the ledger of the given ckERC20 token to approve the minter to burn some of the user's ckERC20 tokens.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+3. The user calls the `withdraw_erc20` function on the ckETH minter, specifying the canister ID of the ledger of the given ckERC20 token, the amount to be withdrawn, and the Ethereum destination address. The ckETH minter then issues transactions to burn the fee in ckETH and the specififed amount of the ckERC20 token. If both operations succeed, it sends an Ethereum transaction that transfers the specified amount from the helper smart contract to the destination address that the user provided.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:62]:
```suggestion
3. The user calls the `withdraw_erc20` function on the ckETH minter, specifying the canister ID of the ledger for the given ckERC20 token, the amount to be withdrawn, and the Ethereum destination address. The ckETH minter then issues transactions to burn the fee in ckETH and the specified amount of the ckERC20 token. If both operations succeed, it sends an Ethereum transaction that transfers the specified amount from the helper smart contract to the destination address that the user provided.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+There are several different ways to interact with ckERC20 tokens:
+
+- Converting ckERC20 to ERC-20 tokens
+- Converting ERC-20 to ckERC20 tokens
+- Transferring ckERC20 tokens
+- Viewing ckERC20 token transactions
+
+## Converting ERC-20 tokens to ckERC20 tokens
+
+First [download the ckETH repository](https://github.com/dfinity/ic/tree/master/rs/ethereum/cketh). Then, navigate into the subdirectory for either the mainnet(ic/rs/ethereum/cketh/mainnet) or testnet(ic/rs/ethereum/cketh/testnet).
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:25]:
```suggestion
First [download the ckETH repository](https://github.com/dfinity/ic/tree/master/rs/ethereum/cketh). Then, navigate into the subdirectory for either the mainnet (ic/rs/ethereum/cketh/mainnet) or testnet (ic/rs/ethereum/cketh/testnet).
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+
+:::note
+
+Note that ERC-20 tokens can only be converted if the ICRC ledger suite for the
+corresponding ckERC20 token exists. The list of supported ckERC20 tokens can be obtained by running the following command:
+
+:::
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:32]:
```suggestion

```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+
+```
+dfx canister --ic call minter get_minter_info
+```
+
+The return value contains a vector `supported_ckerc20_tokens`, which lists all ckERC20 tokens.
+The return value also contains the field `erc20_helper_contract_address`, which states the **helper smart contract** address on Ethereum used in the conversion flows.
+
+Currently, the ckERC20 helper contracts are the following.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:41]:
```suggestion
The ckERC20 helper contract addresses can be found below:
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+
+|Ethereum chain |Helper smart contract |
+|---------------|----------------------|
+|[Ethereum Mainnet](https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-1.json)|[0x6abDA0438307733FC299e9C229FD3cc074bD8cC0](https://etherscan.io/address/0x6abDA0438307733FC299e9C229FD3cc074bD8cC0)|
+|[Ethereum Testnet Sepolia](https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-11155111.json)|[0x674Cdbe64Df412DA9bAb1596e00c1520979B5A23](https://sepolia.etherscan.io/address/0x674Cdbe64Df412DA9bAb1596e00c1520979B5A23)|
+
+:::info
+
+Note that the helper smart contract might change. It is therefore recommended to request the (ckETH) minter info and extract the helper smart contract address from the response.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:50]:
```suggestion
Note that the helper smart contract might change. It is therefore recommended to request the ckETH minter info and extract the helper smart contract address from the response to assure you have the latest contract address.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+
+:::
+
+```
+dfx canister --ic call minter get_minter_info
+```
+
+The return value contains a vector `supported_ckerc20_tokens`, which lists all ckERC20 tokens.
+The return value also contains the field `erc20_helper_contract_address`, which states the **helper smart contract** address on Ethereum used in the conversion flows.
+
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:40]:
```suggestion
:::
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+|[Ethereum Mainnet](https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-1.json)|[0x6abDA0438307733FC299e9C229FD3cc074bD8cC0](https://etherscan.io/address/0x6abDA0438307733FC299e9C229FD3cc074bD8cC0)|
+|[Ethereum Testnet Sepolia](https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-11155111.json)|[0x674Cdbe64Df412DA9bAb1596e00c1520979B5A23](https://sepolia.etherscan.io/address/0x674Cdbe64Df412DA9bAb1596e00c1520979B5A23)|
+
+:::info
+
+Note that the helper smart contract might change. It is therefore recommended to request the (ckETH) minter info and extract the helper smart contract address from the response.
+
+:::
+
+If the targeted ckERC20 token is listed, the first step is to create an Ethereum transaction that calls the `approve` function on the corresponding ERC-20 contract address, allowing the helper smart contract to withdraw some of the user's funds.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:54]:
```suggestion
If the targeted ckERC20 token is listed, the next step is to create an Ethereum transaction that calls the `approve` function on the corresponding ERC-20 contract address, allowing the helper smart contract to withdraw some of the user's funds.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+
+The next step is to call the `deposit` function of the helper smart contract, specifying the ERC-20 smart contract address, the amount to be deposited, and the principal ID that should be credited for the deposit. The principal ID must be encoded as a `bytes32` array. A convenient way to obtain the correct encoding is to use the `Principal → Bytes32 conversion` function on the ckETH minter [dashboard](https://sv3dd-oaaaa-aaaar-qacoa-cai.raw.icp0.io/dashboard).
+
+:::warning
+
+If your encoded principal ID is incorrect, funds will be lost.
+
+:::
+
+No further user action is required. The icrc1_balance_of function can be called on the corresponding ckERC20 ledger to check if the ckERC20 tokens have been minted, which usually takes about 20 minutes.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:64]:
```suggestion
No further user action is required. The `icrc1_balance_of` function can be called on the corresponding ckERC20 ledger to check if the ckERC20 tokens have been minted, which usually takes about 20 minutes.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 20]
```diff
@@ -0,0 +1,120 @@
+---
+keywords: [advanced, ethereum, eth, tutorial, convert eth, convert cketh, transfer cketh, use cketh, send cketh]
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import useBaseUrl from "@docusaurus/useBaseUrl";
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Making transactions
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:10]:
```suggestion
# Making ckERC20 transactions
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx [hunk 21]
```diff
@@ -0,0 +1,102 @@
+---
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:1]: I would suggest adding some examples of current ckERC20 tokens, such as ckUSDC and ckLINK. We've had several users ask about the ckUSDC docs, so we should reference at least ckUSDC somewhere in this overview so we have something to point them to.

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 22]
```diff
@@ -0,0 +1,120 @@
...
+```
+dfx canister --ic call minter withdraw_erc20 "(record { amount = CKERC20_TOKEN_AMOUNT:nat; ckerc20_ledger_id = principal \"CKERC20_LEDGER\"; recipient = \"ETH_ADDRESS\"})"
+```
+
+As before, `CKERC20_TOKEN_AMOUNT` and `CKERC20_LEDGER` must be replaced by the desired amount in the smallest unit and the canister ID of the ckERC20 ledger, respectively.
+Lastly, `ETH_ADDRESS` must be replaced by the desired Ethereum destination address.
+
+## Transferring ckERC20 tokens
+
+ckERC20 tokens are standard ICRC-1 tokens and can be transferred by calling the `ircr1_transfer` function.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:95]:
```suggestion
ckERC20 tokens are standard ICRC-1 tokens and can be transferred by calling the `icrc1_transfer` function.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 23]
```diff
@@ -0,0 +1,120 @@
...
+
+:::note
+
+Transferring ckERC20 tokens incurs a small fee. The exact amount depends on the specific ckERC20 token. The transfer fee can be obtained by calling the `icrc1_metadata` function on the ckERC20 ledger. The response contains the fee under the key `icrc1:fee`.
+
+:::
+
+## Viewing ckERC20 token transactions
+
+The ckERC20 tokens have not been added to the Internet Computer dashboard yet.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:111]: Aren't the ckUSDC transactions on the dashboard? https://dashboard.internetcomputer.org/ethereum/xevnm-gaaaa-aaaar-qafnq-cai

IMO we should include mention of ckUSDC here and the link to the transaction dashboard.

File: docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx (hunk 2 above)
Comment (THLO) [reply docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx:26]: Any fees such as transfer fees or conversion fees (in both directions).

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx (hunk 21 above)
Comment (THLO) [reply docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:1]: Good point. I'll add a short paragraph.

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx (hunk 23 above)
Comment (THLO) [reply docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:111]: You're right. Last time I checked, these transactions were not there yet.   :slightly_smiling_face: 
I'll add a paragraph mentioning concrete ckERC20 tokens including links here.

Comment (THLO) [reply docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx]: Done! Please take another look.

File: docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx [hunk 24]
```diff
@@ -19,6 +19,18 @@ Chain-key tokens can be swapped for one another seamlessly. Making transactions
 
 Each implementation of a chain-key token uses the [ICRC](https://github.com/dfinity/ICRC) token standard so that they can be easily integrated by services such as wallets and DEXs. To implement smart contracts that use chain-key tokens, simply make calls from a smart contract to the ICRC ledger for the chain-key token you wish to use. Using chain-key tokens does not require implementing a specific type of smart contract, and implementation is not restricted to a specific programming language.
 
+## Requirements for chain-key tokens
+
+In addition to the requirement that a chain-key token must adhere to the [ICRC](https://github.com/dfinity/ICRC) token standard, there are additional requirements to ensure a high degree of security and decentralization for all chain-key tokens:
+
+- The chain-key token must be a true digital "twin" in the sense that any user can convert the native token to the chain-key token and vice versa, paying only for the fees. This implies that the chain-key token must be fully backed by the corresponding native token at all times. Conversions may be delayed if one chain is temporarily unavailable but all conversions must be finalized eventually.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx:26]:
```suggestion
- The chain-key token must be a true digital "twin" in the sense that any user can convert the native token to the chain-key token and vice versa, paying only for the transfer and conversion fees. This implies that the chain-key token must be fully backed by the corresponding native token at all times. Conversions may be delayed if one chain is temporarily unavailable but all conversions must be finalized eventually.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx [hunk 25]
```diff
@@ -19,6 +19,18 @@ Chain-key tokens can be swapped for one another seamlessly. Making transactions
...
+
+- All conversions must be transparent. It must be possible to trace all tokens on-chain.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx:28]:
```suggestion
- All conversions must be transparent. All tokens must be traceable on-chain. 
```

File: docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx [hunk 26]
```diff
@@ -19,6 +19,18 @@ Chain-key tokens can be swapped for one another seamlessly. Making transactions
...
+
+- The canister handling the conversions must use chain-key signing capabilities to manage the underlying native token. Moreover, it must read data from the native chain in a decentralized manner.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx:30]:
```suggestion
- The canister handling the conversions must use chain-key signatures to manage the underlying native token. Moreover, it must read data from the native chain in a decentralized manner.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx [hunk 27]
```diff
@@ -19,6 +19,18 @@ Chain-key tokens can be swapped for one another seamlessly. Making transactions
...
+
+- All canisters must be decentralized, meaning all source code must be open, builds must be reproducible, and the canisters must be immutable or controlled by a DAO (for example, an [SNS](https://internetcomputer.org/docs/current/developer-docs/daos/sns) or a [threshold canister](https://github.com/dfinity/threshold) with a large number of verified voters).
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/overview.mdx:32]:
```suggestion
- All canisters must be decentralized, meaning:
    - All source code must be open source.
    - Builds must be reproducible.
    - The canisters must be immutable or controlled by a DAO, for example, an [SNS](https://internetcomputer.org/docs/current/developer-docs/daos/sns) or a [threshold canister](https://github.com/dfinity/threshold) with a large number of verified voters.
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/using-cketh-in-dapps.mdx [hunk 28]
```diff
@@ -0,0 +1,21 @@
...
+
+## Overview
+
+All ckERC20 tokens are ICRC-2 compliant tokens, which means that they support the [ICRC-1](https://github.com/dfinity/ICRC-1/blob/main/standards/ICRC-1/README.md) and [ICRC-2](https://github.com/dfinity/ICRC-1/blob/main/standards/ICRC-2/README.md) token standards.
+
+To integrate ckERC20 tokens into your dapp, you can simply write code that uses the [ICRC-1](https://github.com/dfinity/ICRC-1/blob/main/standards/ICRC-1/README.md) and [ICRC-2](https://github.com/dfinity/ICRC-1/blob/main/standards/ICRC-2/README.md) standards. For token transfers and account balances, requests must be sent to the corresponding ckERC20 ledger canister.
+
+## Write a smart contract that uses ckERC20 tokens
+
+The same code as provided for [ckBTC](/docs/current/developer-docs/multi-chain/chain-key-tokens/ckbtc/using-ckbtc-in-dapps#write-a-smart-contract-that-uses-ckbtc) and [ckETH](/docs/current/developer-docs/multi-chain/chain-key-tokens/cketh/using-cketh-in-dapps#write-a-smart-contract-that-uses-cketh) can be used, the only difference being the targeted ledger canister.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/using-cketh-in-dapps.mdx:21]: Should we include an example showing the ckUSDC ledger canister as the targeted canister?

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx [hunk 29]
```diff
@@ -0,0 +1,108 @@
...
+
+:::info
+
+Note that the [same requirements](/docs/current/developer-docs/multi-chain/chain-key-tokens/overview#requirements-for-chain-key-tokens) as for any other type of chain-key token apply.
+
+:::
+
+## How it works
+
+The ckERC20 functionality is primarily provided by two canisters, the **ckETH minter** and the **ledger suite orchestrator (LSO)**, and a helper smart contract on Ethereum:
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:32]:
```suggestion
The ckERC20 functionality is provided by two canisters, the **ckETH minter** and the **ledger suite orchestrator (LSO)**, and a helper smart contract on Ethereum:
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/using-cketh-in-dapps.mdx (hunk 28 above)
Comment (THLO) [reply docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/using-cketh-in-dapps.mdx]: Good idea. Done!

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx (hunk 17 above)
Comment (THLO) [reply docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:42]: Why is this needed?

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx [hunk 30]
```diff
@@ -0,0 +1,108 @@
...
+
+More details on both flows are provided on the [ckERC20 GitHub page](https://github.com/dfinity/ic/blob/master/rs/ethereum/cketh/docs/ckerc20.adoc), whereas a detailed description of the ledger suite orchestrator can be found in its [GitHub repository](https://github.com/dfinity/ic/tree/master/rs/ethereum/ledger-suite-orchestrator).
+
+The ledger and index canisters in any ckERC20 ledger suite are regular ICRC-1 compliant canisters. A guide on how to interact with these canisters can be found in the [using the ICRC-1 ledger](/docs/developer-docs/defi/icrc-1/using-icrc1-ledger.mdx) section.
+
+## ckERC20 use-cases
+
+ckERC20 tokens can be used in the same way as other chain-key tokens. Use cases include:
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:76]:
```suggestion
ckERC20 tokens can be used in the same way as other chain-key tokens. Use-cases include:
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 31]
```diff
@@ -0,0 +1,126 @@
...
+```
+dfx canister --ic call minter withdraw_erc20 "(record { amount = CKERC20_TOKEN_AMOUNT:nat; ckerc20_ledger_id = principal \"CKERC20_LEDGER\"; recipient = \"ETH_ADDRESS\"})"
+```
+
+As before, `CKERC20_TOKEN_AMOUNT` and `CKERC20_LEDGER` must be replaced by the desired amount in the smallest unit and the canister ID of the ckERC20 ledger, respectively.
+Lastly, `ETH_ADDRESS` must be replaced by the desired Ethereum destination address.
+
+## Transferring ckERC20 tokens
+
+ckERC20 tokens are standard ICRC-1 tokens and can be transferred by calling the `icrc1_transfer` function.
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:97]: >standard ICRC-1 tokens

The other pages refer to ckERC20 as an ICRC-2 compliant token, but then here reference it as a 'standard ICRC-1 token'. This is confusing, is there a reason why here it is specified as an ICRC-1 token and not ICRC-2? Could this be reworded to say that it complies with ICRC-1 and ICRC-2, and therefore can be transferred with the `icrc1_transfer` function?

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx [hunk 32]
```diff
@@ -0,0 +1,126 @@
...
+- [ckLINK transactions](https://dashboard.internetcomputer.org/ethereum/g4tto-rqaaa-aaaar-qageq-cai/transactions)
+
+Alternatively, the history of transactions can be obtained from the index canister of the ckERC20 ledger suite.
+The following command can be used to retrieve the canister IDs of all canisters in the ledger suite of a particular ckERC20 token.
+
+```
+dfx canister --network ic call LSO canister_ids "(record { chain_id = CHAIN_ID:nat; address = "ERC_20_CONTRACT_ADDRESS"})"
+```
+
+`LSO` must be replaced by the canister ID of the ledger suite orchestrator associated with the targeted Ethereum network (for example, it is `vxkom-oyaaa-aaaar-qafda-cai` for Ethereum mainnet), `CHAIN_ID` must be replaced by the chain ID of the targeted Ethereum network (for example, `chain_id = 1` for Ethereum mainnet). Lastly, `ERC_20_CONTRACT_ADDRESS` must be replaced by the corresponding ERC-20 smart contract address (for example, `address = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48"` for USDC).
```
Comment (jessiemongeon1) [inline docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx:126]:
```suggestion

Replace the following values:

- `LSO`: The canister ID of the ledger suite orchestrator associated with the targeted Ethereum network (`vxkom-oyaaa-aaaar-qafda-cai` for Ethereum mainnet).

- `CHAIN_ID`: The chain ID of the targeted Ethereum network (`chain_id = 1` for Ethereum mainnet). 

- `ERC_20_CONTRACT_ADDRESS`: The corresponding ERC-20 smart contract address (`address = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48"` for USDC).
```

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx (hunk 30 above)
Comment (THLO) [reply docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:76]: Isn't "use case" more commonly used? It looks like [Merriam-Webster agrees](https://www.merriam-webster.com/dictionary/use%20case).

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx (hunk 31 above)
Comment (THLO) [reply docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/making-transactions.mdx]: Good point. Done!

File: docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx (hunk 30 above)
Comment (jessiemongeon1) [reply docs/developer-docs/multi-chain/chain-key-tokens/ckerc20/overview.mdx:76]: To match the title 'ckERC20 use-cases' above, it should be 'use-cases'. The other Chain Key docs hyphenate it as well. 

As a separate note, I will look into standardizing it across the docs.

**Review by jessiemongeon1 — APPROVED** [review]

**Pull request merged**